parsons/databases/database/database.py

```diff
@@ -0,0 +1,257 @@
+import parsons.databases.database.constants as consts
+import ast
+
+
+class DatabaseCreateStatement():
+
+    def __init__(self):
+        self.INT_TYPES = consts.INT_TYPES
+        self.SMALLINT = consts.SMALLINT
+        self.SMALLINT_MIN = consts.SMALLINT_MIN
+        self.SMALLINT_MAX = consts.SMALLINT_MAX
+        self.MEDIUMINT = consts.MEDIUMINT
+        self.MEDIUMINT_MIN = consts.MEDIUMINT_MIN
+        self.MEDIUMINT_MAX = consts.MEDIUMINT_MAX
+        self.INT = consts.INT
+        self.INT_MIN = consts.INT_MIN
+        self.INT_MAX = consts.INT_MAX
+        self.BIGINT = consts.BIGINT
+        self.FLOAT = consts.FLOAT
+        self.VARCHAR = consts.VARCHAR
+        self.RESERVED_WORDS = consts.RESERVED_WORDS
+        self.COL_NAME_MAX_LEN = consts.COL_NAME_MAX_LEN
+        self.IS_CASE_SENSITIVE = consts.IS_CASE_SENSITIVE
+
+    # This will allow child classes to modify how these columns are handled.
+    def _rename_reserved_word(self, col, index=None):
```

Comment:
The [Redshift](https://github.com/dannyboy15/parsons/blob/database-refactor/parsons/databases/postgres/postgres_create_statement.py#L24-L36) and [Postgres](https://github.com/dannyboy15/parsons/blob/database-refactor/parsons/databases/postgres/postgres_create_statement.py#L24-L36) classes both use the index for handling reserved words.

```python
return f"col_{index}"
```
Therefore [when it's called](https://github.com/dannyboy15/parsons/blob/database-refactor/parsons/databases/database/database.py#L221) we have to pass in both the column name and the index even it we don't plan on using it. Otherwise we would have to re-implement the whole `format_column` method in a child class.